src/entities/Votes/utils.ts

```diff
@@ -168,16 +170,46 @@ function getNumber(number: number) {
   return Math.floor(number || 0)
 }
 
+async function getScores(addresses: string[], block?: string | number) {
+  addresses = addresses.map((addr) => addr.toLowerCase())
+  const { scores, strategies } = await SnapshotApi.get().getScores(addresses, block)
+
+  const result: DetailedScores = {}
+  for (const addr of addresses) {
+    result[addr] = {
+      ownVp: 0,
+      delegatedVp: Math.round(scores[strategies.findIndex((s) => s.name === DELEGATION_STRATEGY_NAME)][addr]) || 0,
+      totalVp: 0,
+    }
+  }
+
+  for (const score of scores) {
+    for (const addr of Object.keys(score)) {
+      const address = addr.toLowerCase()
+      result[address].totalVp = (result[address].totalVp || 0) + Math.floor(score[addr] || 0)
+    }
+  }
+
+  for (const address of Object.keys(result)) {
+    result[address].ownVp = result[address].totalVp - result[address].delegatedVp
+  }
+
+  return result
+}
+
+export async function getLatestScores(addresses: string[]) {
+  return await getScores(addresses)
+}
+
+export async function getVotingPower(address: string) {
+  const vp = await getLatestScores([address])
```

Comment:
You can call here `getScores([address]` and remove getLatestScores function. Or am I missing something?